cristin-keyword/src/main/java/no/unit/nva/cristin/keyword/query/QueryKeywordsApiClient.java

```diff
@@ -0,0 +1,72 @@
+package no.unit.nva.cristin.keyword.query;
+
+import static java.util.Arrays.asList;
+import static no.unit.nva.HttpClientProvider.defaultHttpClient;
+import static no.unit.nva.utils.UriUtils.createCristinQueryUri;
+import static no.unit.nva.utils.UriUtils.getNvaApiUri;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpResponse;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import no.unit.nva.cristin.common.client.ApiClient;
+import no.unit.nva.cristin.common.client.QueryApiClient;
+import no.unit.nva.cristin.model.CristinTypedLabel;
+import no.unit.nva.cristin.model.SearchResponse;
+import no.unit.nva.model.TypedLabel;
+import nva.commons.apigateway.exceptions.ApiGatewayException;
+import nva.commons.apigateway.exceptions.BadGatewayException;
+import nva.commons.core.paths.UriWrapper;
+
+public class QueryKeywordsApiClient extends ApiClient implements QueryApiClient<Map<String, String>, TypedLabel> {
+
+    public static final String KEYWORD_PATH = "keyword";
+    public static final URI KEYWORD_ID_URI = getNvaApiUri(KEYWORD_PATH);
+    public static final String CRISTIN_KEYWORDS_PATH = "keywords";
+    public static final String KEYWORD_CONTEXT_JSON = "https://example.org/keyword-context.json";
+
+    public QueryKeywordsApiClient() {
+        this(defaultHttpClient());
+    }
+
+    public QueryKeywordsApiClient(HttpClient client) {
+        super(client);
+    }
+
+    @Override
+    public SearchResponse<TypedLabel> executeQuery(Map<String, String> params) throws ApiGatewayException {
+        var convertedParams = new QueryParamConverter(params).convert().getResult();
+        var queryUri = createCristinQueryUri(convertedParams, CRISTIN_KEYWORDS_PATH);
+        var start = System.currentTimeMillis();
+        var response = queryUpstream(queryUri);
+        var keywords = getKeywords(response);
+        var totalProcessingTime = calculateProcessingTime(start, System.currentTimeMillis());
+
+        return new SearchResponse<TypedLabel>(appendSearchStringToId(params))
+                   .withContext(KEYWORD_CONTEXT_JSON)
+                   .withHits(keywords)
+                   .usingHeadersAndQueryParams(response.headers(), params)
+                   .withProcessingTime(totalProcessingTime);
+    }
+
+    private HttpResponse<String> queryUpstream(URI uri) throws ApiGatewayException {
+        var response = fetchQueryResults(uri);
+        checkHttpStatusCode(KEYWORD_ID_URI, response.statusCode(), response.body());
+
+        return response;
+    }
+
+    private List<TypedLabel> getKeywords(HttpResponse<String> response) throws BadGatewayException {
+        var keywords = asList(getDeserializedResponse(response, CristinTypedLabel[].class));
+
+        return keywords.stream()
+                   .map(keyword -> new TypedLabel(keyword.getCode(), keyword.getName()))
```

Comment:
Wouldn't it be better to create a toTypedLabel in CristinTypedLabel?

Reply:
since this was the only place it was used i found it was better to just inline. I could create a mapper class between CristinTypedLabel and TypedLabel but since its only used here for the time being and it just has 2 fields, i decided to inline for now